CMakeLists.txt

```
@@ -58,14 +58,22 @@ set(DIR_PERMISSIONS DIRECTORY_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE G
 # ROOT build. This is significant e,g. for roottest, which will otherwise have libraries
 # of a different ROOT build available / visible / reachable.
 if(NOT $ENV{ROOTSYS} STREQUAL "")
-  string(REPLACE "$ENV{ROOTSYS}/bin" "" ENV_PATH "$ENV{PATH}")
-  string(REPLACE "$ENV{ROOTSYS}/lib" "" ENV_LD_LIBRARY_PATH "$ENV{LD_LIBRARY_PATH}")
-  string(REPLACE "$ENV{ROOTSYS}/lib" "" ENV_PYTHONPATH "$ENV{PYTHONPATH}")
-  string(REPLACE "$ENV{ROOTSYS}" "" ENV_CMAKE_PREFIX_PATH "$ENV{CMAKE_PREFIX_PATH}")
-  set(ENV{PATH} "${ENV_PATH}")
-  set(ENV{LD_LIBRARY_PATH} "${ENV_LD_LIBRARY_PATH}")
-  set(ENV{PYTHONPATH} "${ENV_PYTHONPATH}")
-  set(ENV{CMAKE_PREFIX_PATH} "${ENV_CMAKE_PREFIX_PATH}")
+  function(strip_path ENV_VAR _path)
+    string(REPLACE ":${_path}:" ":" _temp_envvar "${_temp_envvar}")
+    string(REGEX REPLACE "^${_path}:" "" _temp_envvar "$ENV{${ENV_VAR}}")
+    string(REGEX REPLACE ":${_path}$" "" _temp_envvar "${_temp_envvar}")
```

It looks like the lines were reordered, so the last argument needs to change.
And after the first line dealt with `:A:`, you should get away with a single regex:
```suggestion
    string(REPLACE ":${_path}:" ":" _temp_envvar "$ENV{${ENV_VAR}}")
    string(REGEX REPLACE ":?${_path}:?" "" _temp_envvar "${_temp_envvar}")
```

---

> It looks like the lines were reordered, so the last argument needs to change.

Indeed, exactly what happeed :(

---

> And after the first line dealt with :A:, you should get away with a single regex:
```
string(REGEX REPLACE ":?${_path}:?" "" _temp_envvar "${_temp_envvar}")
```
I think they would still wrong partially replace
```
/some/other/path/${_path}
${_path}64
...:${_path}64:...
etc.
```

---

I see. If `A` is not a full path, that could indeed create a problem